### PR TITLE
Create SortCommand feature v2

### LINKED ISSUE #133: Create SortCommand feature
Trying out a SortCommand feature that sorts the names in the addressbook in alphabetical order

### PR DESCRIPTION
**Pull request by BuffWuff1712**
fixes #133.

created an internal ID for each Person to keep track of the order in which each Person Object is created. List command would also sort each Person according to their internal ID, while Sort command would sort each Person by their full name.

only possible flaw with this is that the internal ID is immutable. For eg if I add Person A, B, C and D (with IDs 1,2,3,4 respectively), and I remove Person B, Persons C and D will still have IDs 3 and 4 instead of being updated to 2 and 3.

Sorting will temper with the actual addressbook, as tempering with the UI display only will cause some issue when commands like Show are used right after Sort

### CODE REVIEW

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 1]
```diff
@@ -55,6 +56,8 @@ public class HelpWindow extends UiPart<Stage> {
             + "\n\n"
             + HELP_SHOW_COMMAND
             + "\n\n"
+            + HELP_SORT_COMMAND
```
Comment (tayxuenye) [inline src/main/java/seedu/address/ui/HelpWindow.java:59]: Should probably state that sort is permanent and it sorts by alphabetical order and to unsort, use list command.

**Review by tayxuenye — APPROVED** [review]
Reviewed and tested. Would suggest explaining sort more in-depth in helpWindow (see comment).

**Review by Vatinius — CHANGES_REQUESTED** [review]
Don't have the time to test it fully yet, but I caught the same bug whereby it favoured capitalised names first again. Will test it more later when I have time.

image

**Comment by Tkaixiang** [comment]
> Don't have the time to test it fully yet, but I caught the same bug whereby it favoured capitalised names first again. Will test it more later when I have time.
> 
> image

I think this is because capital letters are earlier on the ASCII table. I think we can consider a case insensitive sort instead and just dont mind if capital comes first or not since users probably wont prioritise that.

**Review by Vatinius — COMMENTED** [review]
Noticed that when i'm in "sorted" view and add a new contact, they are placed at the bottom of the list rather than in the alphabetical order like in previous iterations

**Review by Vatinius — APPROVED** [review]
I shall pre-approve first to enable a smoother process to merge in the event i'm not available to approve it later on.

**Pull request merged**